modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/store/test/DLStoreStoreAreaTest.java

```diff
@@ -0,0 +1,277 @@
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.configuration.test.util.ConfigurationTestUtil;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.AssumeTestRule;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.util.HashMapDictionaryBuilder;
+import com.liferay.portal.kernel.util.PropsKeys;
+import com.liferay.portal.kernel.util.PropsUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.test.rule.FeatureFlags;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+
+/**
+ * @author Adolfo Pérez
+ */
+@FeatureFlags("LPS-174816")
+@RunWith(Arquillian.class)
+public class DLStoreStoreAreaTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new AssumeTestRule("assume"), new LiferayIntegrationTestRule());
+
+	public static void assume() {
+		String gcsStoreClassName = "com.liferay.portal.store.gcs.GCSStore";
+		String dlStoreImpl = PropsUtil.get(PropsKeys.DL_STORE_IMPL);
+
+		Assume.assumeTrue(
+			StringBundler.concat(
+				"Property \"", PropsKeys.DL_STORE_IMPL, "\" is not set to \"",
+				gcsStoreClassName, "\""),
+			dlStoreImpl.equals(gcsStoreClassName));
+	}
+
+	@BeforeClass
+	public static void setUpClass() throws Exception {
+		_configuration = _configurationAdmin.getConfiguration(
+			"com.liferay.portal.store.gcs.configuration.GCSStoreConfiguration",
+			StringPool.QUESTION);
+
+		ConfigurationTestUtil.saveConfiguration(
+			_configuration,
+			HashMapDictionaryBuilder.<String, Object>put(
+				"aes256Key", ""
+			).put(
+				"bucketName", "test"
+			).put(
+				"initialRetryDelay", "400"
+			).put(
+				"initialRPCTimeout", "120000"
+			).put(
+				"maxRetryAttempts", "5"
+			).put(
+				"maxRetryDelay", "10000"
+			).put(
+				"maxRPCTimeout", "600000"
+			).put(
+				"retryDelayMultiplier", "1.5"
+			).put(
+				"retryJitter", "false"
+			).put(
+				"rpcTimeoutMultiplier", "1.0"
+			).put(
+				"serviceAccountKey", ""
+			).build());
+	}
+
+	@Test
+	public void testAddFile() throws Exception {
+		String fileName = StringUtil.randomString();
+
+		_addFile(fileName, Store.VERSION_DEFAULT);
+
+		Assert.assertTrue(
+			DLStoreUtil.hasFile(
+				TestPropsValues.getCompanyId(), TestPropsValues.getGroupId(),
+				fileName, Store.VERSION_DEFAULT));
+
+		StoreArea.withStoreArea(
+			StoreArea.LIVE,
+			() -> _assertHasStoreFile(fileName, Store.VERSION_DEFAULT));
```

Comment:
_Optional_ for next changes maybe

It would interesting also to assert the opposite for the other area, whenever `_assertHasStoreFile` is used. In this case I would assert also the following:

```java
StoreArea.withStoreArea(
			StoreArea.DELETED,
			() -> _assertHasNotStoreFile(fileName, Store.VERSION_DEFAULT));
```